### PR TITLE
update WAMR

### PR DESCRIPTION
**Pull request by qinkunbao**
## Description

1. Update WAMR dependencies.
2. Remove assert and unwrap.

## Type of change (select or add applied and delete the others)

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] API change with a documentation update
- [ ] Additional test coverage
- [ ] Code cleanup or just sync with upstream third-party crates

## Checklist

- [x] Fork the repo and create your branch from `master`.
- [x] If you've added code that should be tested, add tests.
- [x] If you've changed APIs, update the documentation.
- [x] Ensure the tests pass (see CI results).
- [x] Make sure your code lints/format.

### CODE REVIEW

File: executor/src/wamr.rs [hunk 1]
```diff
@@ -100,6 +101,19 @@ extern "C" {
 #[derive(Default)]
 pub struct WAMicroRuntime;
 
+macro_rules! early_terminate {
+    ($cond:expr, $msg:expr, $func:block) => {
+        if !($cond) {
+            reset_thread_context()?;
+            #[allow(unused_unsafe)]
```
Comment (dingelish) [inline executor/src/wamr.rs:108]: consider removal. after this pr, `wasm_runtime_***` will looks like "safe" if it's wrapped in early_terminate

**Comment by mssun** [comment]
I suggest to wrap C APIs into a struct and use `new` and `drop` for resource management.

**Comment by qinkunbao** [comment]
> I suggest to wrap C APIs into a struct and use `new` and `drop` for resource management.

Sounds good! Will do.

File: executor/src/wamr.rs [hunk 2]
```diff
@@ -191,17 +213,33 @@ impl TeaclaveExecutor for WAMicroRuntime {
...
+        );
 
         let exec_env = unsafe { wasm_runtime_create_exec_env(module_instance, DEFAULT_STACK_SIZE) };
-        assert!((exec_env as usize) != 0);
+        early_terminate!(
+            (exec_env as usize) != 0,
+            "wasm_runtime_create_exec_env failed",
+            {
+                wasm_runtime_deinstantiate(module_instance);
+                wasm_runtime_unload(module);
```
Comment (bigdata-memory) [inline executor/src/wamr.rs:232]: Looks like this block of code can be further abstracted away w.r.t. DRY.

**Comment by mssun** [comment]
This PR becomes stale. Closing.

**Pull request closed without merging**